mlapi/discriminating_algo.py

```diff
@@ -0,0 +1,129 @@
+import statistics
+
+class DiscriminatingAlgo(object):
+    # Discrimination parameters
+    facets_sample_count = 3  # minimum docs by facet to be accepted
+    tolerance_interval = 10 # max standard deviation accepted ()
+    nbr_facet_values = 2 # to consider a facet as having a large number of values
+
+    facets = [] # names
+    step2 = {}
+    step3 = {}
+
+    def discriminating_facets_algo(self, facets_by_document):
+        documents_by_facet = self.inverse_dictionary(facets_by_document)
+        data = self.get_formated_data(documents_by_facet)
+        self.execute_algorithm(data)
+        if not (len(self.step3)):
```

Comment:
Optional but in Python we usually see "if self.step3" which is equivalent to this condition.

When you're not sure what's the best way, google the "pythonic" way to do X and you'll often find simple and elegant solutions

https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty